src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -309,8 +310,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 auto contextArgs = winrt::make<ContextMenuRequestedEventArgs>(til::point{ pixelPosition }.to_winrt_point());
                 ContextMenuRequested.raise(*this, contextArgs);
             }
-            else
+            else if (!thisClickFocused)
             {
+                // GH#15803: Don't do this if this click activated our terminal
+                // window. We don't want right-click activations of the window
+                // to immediately paste the clipboard.

```

Comment:
Wouldn't it be much better long term if we instead merged `ControlInteractivity` back into `TermControl`?

Reply:
I strongly disagree that it would be better. In another branch in this repo, the WPF control works on top of ControlInteractivity and is unaware of Windows.UI.Xaml.